tests/libnode/node.go

```diff
@@ -122,56 +123,94 @@ func GetNodeDrainKey() string {
+			delete(m, key)
+			return m
+		}
+	}
+
+	Expect(fetchMap).ToNot(BeNil())
+	Expect(mutateMap).ToNot(BeNil())
+
 	virtCli, err := kubecli.GetKubevirtClient()
 	util.PanicOnError(err)
 
 	origNode, err := virtCli.CoreV1().Nodes().Get(context.Background(), nodeName, k8smetav1.GetOptions{})
 	Expect(err).ToNot(HaveOccurred())
 
-	newNode := origNode.DeepCopy()
-	if isLabel {
-		newNode.Labels[key] = value
-	} else {
-		newNode.Annotations[key] = value
+	originalMap := fetchMap(origNode)
+	jsonPath := "/metadata/" + string(mapType) + "s"
+
+	expectedMap := make(map[string]string, len(originalMap))
+	for k, v := range originalMap {
+		expectedMap[k] = v
+	}
+
+	expectedMap = mutateMap(key, value, expectedMap)
+
+	if equality.Semantic.DeepEqual(originalMap, expectedMap) {
+		// key and value already exist in node
+		return
 	}
 
-	// This is done in an inefficient way since we can patch only labels/annotations here.
-	err = nodes.PatchNode(virtCli, origNode, newNode)
+	patch := generateStrMapReplacePatch(jsonPath, originalMap, expectedMap)
+	patchedNode, err := virtCli.CoreV1().Nodes().Patch(context.Background(), nodeName, types.JSONPatchType, patch, k8smetav1.PatchOptions{})
 	Expect(err).ShouldNot(HaveOccurred())
+
+	resultMap := fetchMap(patchedNode)
+
+	const errPattern = "adding %s (key: %s. value: %s) to node %s failed. Expected %ss: %v, actual: %v"
+	Expect(equality.Semantic.DeepEqual(resultMap, expectedMap)).To(BeTrue(), fmt.Sprintf(errPattern, string(mapType), key, value, nodeName, string(mapType), expectedMap, resultMap))
 }
 
 func AddLabelToNode(nodeName, key, value string) {
-	addLabelAnnotationHelper(nodeName, key, value, true)
+	addRemoveLabelAnnotationHelper(nodeName, key, value, label, add)
 }
 
 func AddAnnotationToNode(nodeName, key, value string) {
-	addLabelAnnotationHelper(nodeName, key, value, false)
+	addRemoveLabelAnnotationHelper(nodeName, key, value, annotation, add)
 }
 
 func RemoveLabelFromNode(nodeName string, key string) {
-	virtCli, err := kubecli.GetKubevirtClient()
-	util.PanicOnError(err)
-	node, err := virtCli.CoreV1().Nodes().Get(context.Background(), nodeName, k8smetav1.GetOptions{})
-	Expect(err).ToNot(HaveOccurred())
-
-	if _, exists := node.Labels[key]; !exists {
-		return
-	}
-
-	old, err := json.Marshal(node)
-	Expect(err).ToNot(HaveOccurred())
-	new := node.DeepCopy()
-	delete(new.Labels, key)
-
-	newJson, err := json.Marshal(new)
-	Expect(err).ToNot(HaveOccurred())
-
-	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, node)
-	Expect(err).ToNot(HaveOccurred())
+	addRemoveLabelAnnotationHelper(nodeName, key, "", label, remove)
+}
 
-	_, err = virtCli.CoreV1().Nodes().Patch(context.Background(), node.Name, types.StrategicMergePatchType, patch, k8smetav1.PatchOptions{})
-	Expect(err).ToNot(HaveOccurred())
+func RemoveAnnotationFromNode(nodeName string, key string) {
+	addRemoveLabelAnnotationHelper(nodeName, key, "", label, remove)
```

Comment:
should be `addRemoveLabelAnnotationHelper(nodeName, key, "", annotation, remove)`